hedera-node/hedera-app/src/main/java/com/hedera/node/app/tss/handlers/TssEncryptionKeyHandler.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.tss.handlers;
+
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.hapi.node.state.common.EntityNumber;
+import com.hedera.hapi.node.state.tss.TssEncryptionKeys;
+import com.hedera.hapi.node.transaction.TransactionBody;
+import com.hedera.node.app.spi.workflows.HandleContext;
+import com.hedera.node.app.spi.workflows.HandleException;
+import com.hedera.node.app.spi.workflows.PreCheckException;
+import com.hedera.node.app.spi.workflows.PreHandleContext;
+import com.hedera.node.app.spi.workflows.TransactionHandler;
+import com.hedera.node.app.tss.stores.WritableTssStore;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+/**
+ * Handles the TSS encryption key transaction.
+ */
+@Singleton
+public class TssEncryptionKeyHandler implements TransactionHandler {
+
+    /**
+     * Constructs a {@link TssEncryptionKeyHandler}.
+     */
+    @Inject
+    public TssEncryptionKeyHandler() {}
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void preHandle(@NonNull PreHandleContext context) throws PreCheckException {
+        requireNonNull(context);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckException {
+        requireNonNull(txn);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void handle(@NonNull final HandleContext context) throws HandleException {
+        requireNonNull(context);
+        final var op = context.body().tssEncryptionKeyOrThrow();
+        final var tssStore = context.storeFactory().writableStore(WritableTssStore.class);
+        final var nodeEntityNumber =
+                EntityNumber.newBuilder().number(context.creatorInfo().nodeId()).build();
+        TssEncryptionKeys currentTssEncryptionKeys =
+                tssStore.getTssEncryptionKeys(context.creatorInfo().nodeId());
+        TssEncryptionKeys newTssEncryptionKeys;
+        if (currentTssEncryptionKeys == null) {
```

Comment:
Is it possible the currentEncryptionKey is `Bytes.EMPTY` and we still submit the encryption key transaction. If so, we should set the `currentEncryptionKey`. So this condition should be as below right ?
```
if (currentTssEncryptionKeys == null || currentTssEncryptionKeys.currentEncryptionKey().equals(Bytes.EMPTY)) {
```

Reply:
I don't think that will happen however it makes sense to add this additional condition.